packages/website/private/contents/post-007.mdx

```diff
@@ -0,0 +1,64 @@
+---
+slug: progressive-forms
+title: Progressive forms
+description: Forms work without javascript!
+author: tyler
+date: 2025/02/01
+---
+
+With Oracle clinging on to the trademark for [javascript](https://javascript.tm/), you'll be happy to know that your forms will work without it!
+
+Jokes aside, this is a really cool framework feature that connects server actions directly to your forms.
+
+## What does this look like?
+
+So before React 19 and before we had waku server actions, a very basic form might look like this:
+
+```jsx
+const ClientForm = () => {
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    const formData = new FormData(e.target);
+    await fetch('/api/submit', {
+      method: 'POST',
+      body: JSON.stringify(Object.fromEntries(formData)),
+    });
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <input name="email" type="email" required />
+      <input name="password" type="password" required />
+      <button type="submit">Submit</button>
+    </form>
+  );
+};
+```
+
+This is just fine, but if we want to run our form in an environment with no javascript. Or we just want to optimize the form to not need to render, we can upgrade to a new way.
+
+## Introducing progressive forms with Waku Server actions
+
+In this example, it can be pure RSC built. There is no client JS needed.
```

Comment:
To clarify, the primary goal isn't to support non-JS browser.
Even if you enable JS, the progressive form works while loading JS code and before hydration finishes.

Reply:
reworded this part... let me know what you think :)